tests/test_time_series_graph.py

```diff
@@ -0,0 +1,450 @@
+        self.ground_truth_minimal_graph_1.add_edge('X3 lag(n=2)', 'X2', edge_type=EDGE_T.DIRECTED_EDGE)
+
+        self.tsdag_2 = TimeSeriesCausalGraph()
+        self.tsdag_2.add_nodes_from(['X1'])
+        self.tsdag_2.add_edge('X1 lag(n=2)', 'X1', edge_type=EDGE_T.DIRECTED_EDGE)
+
+        # third graph
+        causal_graph_3 = CausalGraph()
+        nodes = [
+            'A',
+            'B',
+            'C',
+            'D',
+            'E',
+            'F',
+            'A lag(n=1)',
+            'B lag(n=1)',
+            'F lag(n=1)',
+        ]
+        causal_graph_3.add_nodes_from(nodes)
+        causal_graph_3.add_edge('A', 'C')
+        causal_graph_3.add_edge('B', 'C')
+        causal_graph_3.add_edge('C', 'D')
+        causal_graph_3.add_edge('D', 'E')
+        causal_graph_3.add_edge('C', 'E')
+        causal_graph_3.add_edge('E', 'F')
+
+        causal_graph_3.add_edge('A lag(n=1)', 'A')
+        causal_graph_3.add_edge('B lag(n=1)', 'B')
+        causal_graph_3.add_edge('F lag(n=1)', 'F')
+
+        self.ground_truth_minimal_graph_3 = TimeSeriesCausalGraph.from_causal_graph(causal_graph_3.copy())
+
+        causal_graph_3.add_node('C lag(n=1)')
+        causal_graph_3.add_node('D lag(n=1)')
+        causal_graph_3.add_node('E lag(n=1)')
+
+        # we would not need to do this but we want to test that the underlying
+        # time series graph is being correctly constructed
+        for edge in causal_graph_3.edges:
+            source_name = edge.source.identifier
+            dest_name = edge.destination.identifier
+            if 'lag(n=1)' not in source_name and 'lag(n=1)' not in dest_name:
+                causal_graph_3.add_edge(f'{source_name} lag(n=1)', f'{dest_name} lag(n=1)')
+
+        self.tsdag_3 = TimeSeriesCausalGraph.from_causal_graph(causal_graph_3)
+
+    def test_time_series_nodes(self):
+        tsnode = TimeSeriesNode(variable_name='X1', time_lag=-1)
+        tsnode_1 = TimeSeriesNode('X2 lag(n=1)')
+
+        self.assertEqual(tsnode.time_lag, -1)
+        self.assertEqual(tsnode_1.time_lag, -1)
+
+        # test with future lag
+        tsnode = TimeSeriesNode(variable_name='X1', time_lag=3)
+        tsnode_1 = TimeSeriesNode('X2 future(n=3)')
+        self.assertEqual(tsnode.time_lag, 3)
+        self.assertEqual(tsnode_1.time_lag, 3)
+
+    def test_add_node(self):
+        ts_cg = TimeSeriesCausalGraph()
+        ts_cg.add_node('X1 lag(n=1)')
+        assert ts_cg.get_node('X1 lag(n=1)').time_lag == -1
+
+        ts_cg.add_node(variable_name='X1', time_lag=-2)
+        assert ts_cg.get_node('X1 lag(n=2)').time_lag == -2
+
+    def test_add_edge(self):
+        ts_cg = TimeSeriesCausalGraph()
+        ts_cg.add_edge('X1 lag(n=1)', 'X2 lag(n=1)', edge_type=EDGE_T.DIRECTED_EDGE)
+
+        # test that nodes are added correctly
+        assert ts_cg.get_node('X1 lag(n=1)').time_lag == -1
+        assert ts_cg.get_node('X2 lag(n=1)').time_lag == -1
```

Comment:
Same here. In fact, everywhere in this file where simply `assert ...` is used, we should use `self.assertEqual(...)`

Reply:
Will stop commenting on that now, but please change everywhere